python/src/viewer.py

```diff
@@ -555,6 +597,18 @@ def __init__(
                 if len(drawable.getLegend()) > 0:
                     legend_handles.append(artists[0])
                     legend_labels.append(drawable.getLegend())
+                if contour.getColorBarPosition() and len(contour.getLevels()) != 1:
+                    if matplotlib.__version__ >= "3.7.0":
+                        format = None
+                        if contour.getNorm() != "linear" and contour.getLevels():
```

Comment:
I do not understand precisely what means the second part of the expression `and contour.getLevels()`. Does it mean `and len(contour.getLevels()) != 0`?